sklearn/preprocessing/_polynomial.py

```diff
@@ -835,17 +876,33 @@ def fit(self, X, y=None, sample_weight=None):
             accept_sparse=False,
             ensure_min_samples=2,
             ensure_2d=True,
+            force_all_finite=False,
         )
         if sample_weight is not None:
             sample_weight = _check_sample_weight(sample_weight, X, dtype=X.dtype)
 
         _, n_features = X.shape
 
+        if self.handle_missing == "error":
+            if np.isnan(X).any():
+                raise ValueError(
+                    "'X' contains np.nan values, which is conflicting with"
+                    " handle_missing='error'."
+                )
+        else:  # handle_missing == "indicator"
+            self._missing_mask = _get_mask(X, np.nan)
```

Comment:
We should not store this on the estimator, but only use it build the indicator features in a `fit_transform` call.

In the `transform` call, the `X` passed to transform (typically `X_test`) is distinct from the `X` passed to a previous `fit` or `fit_transform` call, hence it should be recomputed.